theblues/plans.py

```diff
@@ -54,18 +60,196 @@ def get_plans(self, reference):
                     "%Y-%m-%dT%H:%M:%SZ"
                 ),
                 description=plan.get('description'),
-                price=plan.get('price')), json))
+                price=plan.get('price')), response))
         except (KeyError, TypeError, ValueError) as err:
-            log.info(
+            log.error(
                 'cannot process plans: invalid JSON response: {!r}'.format(
-                    json))
+                    response))
             raise ServerError(
                 'unable to get list of plans for {}: {}'.format(
                     reference.path(), err))
         except Exception as exc:
-            log.info(
+            log.error(
                 'cannot process plans: invalid JSON response: {!r}'.format(
-                    json))
+                    response))
             raise ServerError(
                 'unable to get list of plans for {}: {}'.format(
                     reference.path(), exc))
+
+    def list_wallets(self):
+        """Get the list of wallets
+
+        @return an object containing a list of wallets, a total, and available
+            credit.
+        @raise ServerError
+        """
+        response = make_request(
+            '{}wallet'.format(self.url),
+            timeout=self.timeout,
+            client=self._client)
+        try:
+            response['wallets'] = tuple(map(
```

Comment:
Why mutating the response rather than creating a brand new dict that we can more easily control?

Reply:
For instance, this could be:
```python
try:
    return {
        'wallets': tuple(Wallet(owner=i['owners], ...) for i in response['wallets']),
        'total': ...
    }
except ...
```